### PR TITLE
Allow VERSION_NAME to be used as name for publishing artifacts

### PR DESCRIPTION
**Pull request by tobrun**
This PR allows VERSION_NAME to be used when publishing artifacts. This is to drive alignment across Mapbox SDK build systems and allow more easily to build binaries on demand.

### CODE REVIEW

File: gradle/artifact-settings.gradle [hunk 1]
```diff
@@ -10,5 +10,5 @@ ext {
     mapboxArtifactScmUrl = 'scm:git@github.com:mapbox/mapbox-navigation-android.git'
     mapboxArtifactLicenseName = 'Mapbox Terms of Service'
     mapboxArtifactLicenseUrl = 'https://www.mapbox.com/legal/tos/'
-    versionName = System.getenv('POM_VERSION_NAME') ?: project.property('POM_SNAPSHOT_VERSION_NAME')
+    versionName = System.getenv('POM_VERSION_NAME') ?: project.hasProperty('VERSION_NAME') ? project.property('VERSION_NAME') : project.property('POM_SNAPSHOT_VERSION_NAME')
```
Comment (LukasPaczos) [inline gradle/artifact-settings.gradle:13]: Could you instead refactor `config.yml` to change the setup to `VERSION_NAME` and drop the duplication?

Comment (tobrun) [reply gradle/artifact-settings.gradle:13]: sure! wanted to make the change as small as possible but happy to rework fully.

File: .circleci/config.yml [hunk 2]
```diff
@@ -295,12 +295,11 @@ commands:
           command: |
             if [[ -n "$CIRCLE_TAG" ]]; then
               if [[ $CIRCLE_TAG == v* ]]; then
-                POM_VERSION_NAME=`echo $CIRCLE_TAG | sed s/v//`
+                sed -i -e "s/^VERSION_NAME=.*/VERSION_NAME=${CIRCLE_TAG:1}/" gradle.properties
```
Comment (LukasPaczos) [inline .circleci/config.yml:298]: There's no `VERSION_NAME` in `gradle.properties` as things stand, so this `sed` doesn't do anything, does it?

File: gradle.properties [hunk 3]
```diff
@@ -11,7 +11,7 @@ POM_LICENCE_DIST=repo
 POM_DEVELOPER_ID=mapbox
 POM_DEVELOPER_NAME=Mapbox
 
-POM_SNAPSHOT_VERSION_NAME=2.4.0-SNAPSHOT
+SNAPSHOT_VERSION_NAME=2.4.0-SNAPSHOT
```
Comment (LukasPaczos) [inline gradle.properties:14]:
```suggestion
VERSION_NAME=2.4.0-SNAPSHOT
```
To resolve the issue with `sed`.

File: gradle/artifact-settings.gradle [hunk 4]
```diff
@@ -10,5 +10,5 @@ ext {
     mapboxArtifactScmUrl = 'scm:git@github.com:mapbox/mapbox-navigation-android.git'
     mapboxArtifactLicenseName = 'Mapbox Terms of Service'
     mapboxArtifactLicenseUrl = 'https://www.mapbox.com/legal/tos/'
-    versionName = System.getenv('POM_VERSION_NAME') ?: project.property('POM_SNAPSHOT_VERSION_NAME')
+    versionName = project.hasProperty('VERSION_NAME') ? project.property('VERSION_NAME') : project.property('SNAPSHOT_VERSION_NAME')
```
Comment (LukasPaczos) [inline gradle/artifact-settings.gradle:13]:
```suggestion
    versionName = project.hasProperty('VERSION_NAME') ? project.property('VERSION_NAME') : System.getenv('VERSION_NAME')
```
To resolve the issue with `sed`.

**Review by LukasPaczos — APPROVED** [review]
LGTM

**Pull request merged**